src/common/Web3Client.ts

```diff
@@ -67,15 +61,18 @@ export default class Web3Client {
       to: _options.to || null,
       data: _options.data,
       encodeAbi: _options.encodeAbi || false,
+      onTransactionHash: _options.onTransactionHash || null,
+      onReceipt: _options.onReceipt || null,
+      onError: _options.onError || null,
     }
   }
 
-  wrapWeb3Promise(promise) {
+  wrapWeb3Promise(promise, options) {
     const _emptyFunc = () => {}
     return promise
-      .on('transactionHash', this.events.onTransactionHash || _emptyFunc)
-      .on('receipt', this.events.onReceipt || _emptyFunc)
-      .on('error', this.events.onError || _emptyFunc)
+      .on('transactionHash', options.onTransactionHash || _emptyFunc)
+      .on('receipt', options.onReceipt || _emptyFunc)
+      .on('error', options.onError || _emptyFunc)
```

Comment:
Just FYI - we don't have to pass options for `on` methods since `promise` is web3 promise only and it will work directly from outside:

```js
await matic.transferMaticEth(to, amount, {from, parent: false, gas: 100000})
    .on('transactionHash', (txHash) => {
       console.log("transactionHahs", txHash)
    })
```

But, we will keep `wrapWeb3Promise` for now for backward compatibility and remove it from future. 

In our docs, we should use demonstrate direct promise instead of passing callbacks in options.

@saylip @rahuldamodar94

Reply:
we are returning async functions instead of web3 promise so currently `maticjs.function().on("transactionHash", (txHash) => { store here })` is not possible

checkout https://github.com/ethereum/web3.js/issues/1547 for more details

Reply:
Makes sense